_posts/fr/2021-09-29-injection-des-dependances.md

```diff
@@ -0,0 +1,345 @@
+---
+layout: post
+title: "L'injection des dépendances dans Symfony"
+excerpt: Vous travaillez avec Symfony, mais l'injection des dépendances reste un peu flou pour vous ? Décovrez son fonctionnement et apprenez comment en tirer profit au maximum.
+authors:
+    - marishka
+permalink: /fr/injection-des-dependances/
+categories:
+    - bonnes pratiques
+    - symfony
+    - di
+
+---
+
+## Injection de dépendances
+L'injection de dépendances est un mécanisme qui permet d'implémenter le principe de l'inversion de contrôle, un des principes [SOLID](https://www.digitalocean.com/community/conceptual_articles/s-o-l-i-d-the-first-five-principles-of-object-oriented-design).
+L'idée est de créer dynamiquement (_injecter_) les dépendances d'une classe en utilisant une description (un fichier de configuration par exemple).
+Cette méthode va nous permettre de ne plus exprimer les dépendances entre les composants dans le code de manière statique, mais de les déterminer dynamiquement à l'exécution.
+
+Prenenos un exemple pour illustrer. Imaginons que nous avons une classe _A_, qui dépend des classes _B_ et _C_.
+Dans mon code, j'aurai besoin de faire ceci :
+
+```php
+<?php
+
+namespace App\Services;
+
+class B implements InterfaceB {
+    // ...
+}
+```
+
+```php
+<?php
+
+namespace App\Services;
+
+class C implements InterfaceC {
+    // ...
+}
+```
+
+```php
+<?php
+
+namespace App\Services;
+
+class A {
+    private B $b;
+    private C $c;
+
+    public function __construct()
+    {
+        $this->b = new B();
+        $this->c = new C();
+    }
+
+    // ...
+}
+```
+
+Avec l'injection des dépendances, je ne vais plus avoir besoin de créer les instances des classes _B_ et _C_ manuellement, mais je vais les injecter :
+
+```php
+<?php
+
+namespace App\Services;
+
+class A {
+    private InterfaceB $b;
+    private InterfaceC $c;
+
+    public function __construct(InterfaceB $b, InterfaceC $c)
+    {
+        $this->b = $b;
+        $this->c = $c;
```

Comment:
Effectivement c'est plus lisible pour comprendre le principe de mettre le $this->b = $b, mais avec PHP8 y a plus besoin.
Peut-être mettre un petit commentaire sur quelle version de PHP tu es